### PR TITLE
Week 3 PR #2 based on review comments

### PR DESCRIPTION
**Pull request by rwackes**
In this PR, I validate client-side input and clarify code structure based on comments left by my hosts on my Week 3 PR.
- I added a required tag to the fields of my form that I need my user to fill in. If a user leaves a required field empty and attempts to submit the comment form, it is not accepted by my application and the user is given an alert to fill out the empty required field. Screenshot: https://screenshot.googleplex.com/vZHnq6QbsH6
- I specified the type of input I am expecting so that if a user does not fill in a field in the expected format, it is not accepted by my application and they are given an alert to fill out the field in the correct format.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,15 +14,15 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.FetchOptions;
```
Comment (vkkpng-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:17]: Nit: alphabetize the import

**Review by vkkpng-g — APPROVED** [review]
Thank you for taking the time to putting the changes in a new pr...it probably wasn't trivial to make this PR so I appreciate the PR!

Looks good to me :)

**Review by wdi-gg — COMMENTED** [review]
Thanks for creating new PR!
Can you point me to the code that validates client-side input and explain which cases are valid / invalid for your website?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (rwackes) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:17]: Oops I accidentally left that. Fixed it by putting it below Entity import line

**Comment by rwackes** [comment]
The code that validates client-side input is in the comments.html. On lines 27, 30, 33, 34, and 39, I added "required" and changed to type=email for email input that requires users write valid data in those input fields. So, until those inputs are filled in, they are highlighted in red (meaning they are required and have invalid input). Until all required input fields are filled in (contact number is not required and can be left blank), if a user attempts to click submit button, it will not post their comment and a HTML drop down tag will appear the first input field that is invalid (from top to bottom). The type=email tag requires a user write text in the format of johnsmith@emailservice.

**Pull request merged**